ddtrace/contrib/botocore/patch.py

```diff
@@ -52,23 +67,155 @@ def inject_trace_data_to_message_attributes(trace_data, entry):
+    :span: the span which provides the trace context to be propagated
+
+    Inject trace headers into MessageAttributes for all SQS or SNS records inside a batch
+    """
     trace_data = {}
     HTTPPropagator.inject(span.context, trace_data)
     params = args[1]
 
-    for entry in params["Entries"]:
+    # An entry here is an SNS or SQS record, and depending on how it was published,
+    # it could either show up under Entries (in case of Publish, PutRecord & PutRecords),
+    # or PublishBatchRequestEntries (in case of PublishBatch). The order doesn't really matter here,
+    # but Entries is more likely than PublishBatchRequestEntries so it makes sense to check there first.
+    entries = params.get("Entries", params.get("PublishBatchRequestEntries", []))
+    for entry in entries:
         inject_trace_data_to_message_attributes(trace_data, entry)
 
 
-def inject_trace_to_sqs_message(args, span):
+def inject_trace_to_sqs_or_sns_message(args, span):
+    # type: (List[Any], Span) -> None
+    """
+    :args: contains args for the current botocore action, SNS/SQS record is at index 1
+    :span: the span which provides the trace context to be propagated
+
+    Inject trace headers into MessageAttributes for the SQS or SNS record
+    """
     trace_data = {}
     HTTPPropagator.inject(span.context, trace_data)
     params = args[1]
 
     inject_trace_data_to_message_attributes(trace_data, params)
 
 
+def inject_trace_to_eventbridge_detail(args, span):
+    # type: (List[Any], Span) -> None
+    """
+    :args: contains args for the current botocore action, EventBridge records are at index 1
+    :span: the span which provides the trace context to be propagated
+
+    Inject trace headers into the EventBridge record if the record's Detail object contains a JSON string
+    Max size per event is 256kb (https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-putevent-size.html)
+    """
+    params = args[1]
+    if "Entries" not in params:
+        log.debug("Unable to inject context. The Event Bridge event had no Entries.")
+        return
+
+    for entry in params["Entries"]:
+        if "Detail" not in entry:
+            detail = {}
+            HTTPPropagator.inject(span.context, detail)
+            entry["Detail"] = json.dumps({"_datadog": detail})
+            continue
+
+        try:
+            data_size = sys.getsizeof(entry["Detail"])
+            if data_size + 512 >= 256000:
+                log.debug("Inject trace context will make Detail exceed max allowed size")
+                return
+
+            detail = json.loads(entry["Detail"])
+            dd_context = detail.get("_datadog", {})
+            HTTPPropagator.inject(span.context, dd_context)
+            detail["_datadog"] = dd_context
+            entry["Detail"] = json.dumps(detail)
+        except Exception:
+            log.warning("Unable to parse Detail and inject span context")
```

Comment:
When would this fail? `Exception` is fairly permissive, we should try to be more specific if we can. Or, we should prefer to raise an exception instead of "silently" failing.